### PR TITLE
tests: Add tests to Task APIs

### LINKED ISSUE #987: Tests: Cover all test cases in task.py file
## Description
Cover all missing tests in `app/api/dao/tasks.py` file
## Test Cases
These two test cases are missing:-
`Mentorship Relation Does not exist` for Delete Task
`Mentorship Relation Does not exist` for Complete Task

## Definition of Done
- [ ] At least one test case tested.
- [x] Approval by 1 mentor.

## Estimation
2 hours

### PR DESCRIPTION
**Pull request by devkapilbansal**
### Description

Added tests for mentorship relation does not exist in Complete Task and Delete Task Tests

Resolves #987

### Type of Change:

- Quality Assurance

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

Run `python -m unittest discover tests` to ensure all test cases passes

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**

- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by jalajcodes — DISMISSED** [review]
LGTM 🎉

**Comment by devkapilbansal** [comment]
@vj-codes @isabelcosta please review this

**Comment by vj-codes** [comment]
@devkapilbansal I'm not good with tests so I have a ques which may sound silly but here's it -
Isn't being in a mentorship relation the only condition under which mentor/mentee can create a task so ,
If there's no mentorship relation , there cannot be any task and complete or delete is certainly not possible .
Is this what you are adding a test for or am I thinking the wrong way ? Please elaborate 😄

**Comment by devkapilbansal** [comment]
@vj-codes yes there cannot be any task without mentorship relation. But it may be possible that due to some issues, wrong task id (id of a task that exists) is passed with wrong mentorship relation and we need to cover it. 
I added these tests to increase test coverage.  However, if it is possible that this condition will not occur at any cost then this should be removed from the file itself where this message is returned.

**Comment by devkapilbansal** [comment]
@isabelcosta can you please review it

**Review by epicadk — DISMISSED** [review]
👍🏼 looks good

File: tests/tasks/test_dao_complete_task.py [hunk 1]
```diff
@@ -37,6 +38,19 @@ def test_achieve_not_existent_task(self):
 
         self.assertEqual(expected_response, actual_response)
 
+    def test_achieve_task_not_in_relation(self):
+        task_id = 1
+
+        self.assertFalse(self.tasks_list_2.find_task_by_id(task_id).get("is_done"))
+        expected_response = (
+            messages.MENTORSHIP_RELATION_DOES_NOT_EXIST,
```
Comment (isabelcosta) [inline tests/tasks/test_dao_complete_task.py:50]: this test is about achieving task that does not exist in relation, but the message is about mentorship relation not existing 🤔 @devkapilbansal

File: tests/tasks/test_dao_complete_task.py [hunk 2]
```diff
@@ -37,6 +38,19 @@ def test_achieve_not_existent_task(self):
...
+            HTTPStatus.NOT_FOUND,
+        )
+
+        actual_response = TaskDAO.complete_task(self.first_user.id, 123123, task_id)
```
Comment (isabelcosta) [inline tests/tasks/test_dao_complete_task.py]: also could we be a little more explicit about what arguments we are calling. I had to check the code to understand what the middle argument was. I know most of the code does not do this, but its something we can improve for sure.
```
TaskDAO.complete_task(
    user_id=self.first_user.id,
    mentorship_relation_id=123123,
    task_id=task_id
)
```

@devkapilbansal let me know what you think

File: tests/tasks/test_dao_complete_task.py (hunk 1 above)
Comment (devkapilbansal) [reply tests/tasks/test_dao_complete_task.py:50]: Yes, @isabelcosta This test checks if a mentorship relation exists or not before trying to delete a task.
Is something missing or wrong here. :confused:

Comment (isabelcosta) [reply tests/tasks/test_dao_complete_task.py:50]: Ah ok, so the name of the test could be improved to reflect that.
`test_delete_task_from_non_existing_relation` something like that 🤔

Comment (devkapilbansal) [reply tests/tasks/test_dao_complete_task.py:50]: Done @isabelcosta

**Comment by devkapilbansal** [comment]
@jalajcodes @epicadk can you please review it again

**Review by isabelcosta — APPROVED** [review]
Awesome! Also, this won't need "need testing" label, since this is a test itself :)
Thank you @devkapilbansal for doing this!

**Review by jalajcodes — APPROVED** [review]
All the tests are passing. 🎉

**Comment by devkapilbansal** [comment]
Looks like I miss one case 🤦🏻‍♂️
Will add it in documentation so that we can take actions on it later

**Pull request merged**